inspirehep/modules/workflows/utils.py

```diff
@@ -141,3 +145,67 @@ def _try_to_log(logfn, *args, **kwargs):
         return res
 
     return _decorator
+
+
+@with_debug_logging
+def get_pdf_in_workflow(obj):
+    """Return the fullpath to the PDF attached to a workflow object
+
+    Return the PDF attached to a workflow object. If no PDF is already attached
+    and the workflow object is related to an arXiv record, fetches on the fly
+    the PDF from arXiv and attaches it.
+    """
+    for filename in obj.files.keys:
+        if filename.endswith('.pdf'):
```

Comment:
Good point. Just bare in mind that this is not where we download the PDF. This check here already assumes a PDF has been downloaded (thus making `get_pdf_in_workflow` idempotent).

There are two real situations: either the PDF is going to be downloaded proactively from arXiv. In that case we can trust the PDF is a PDF, either it is provided as URL by the user. Indeed this is risky and my code doesn't improve nor worsen the existing code. 
The line you mention is real to find if we indeed have already downloaded something that we think being a PDF.

I can add in a subsequent commit to ensure that what we fetch is actually a PDF. (and as @david-caro  mentioned IRL, that is not larger than a given large enough size).